src/corefx/System.Globalization.Native/locale.cpp

```diff
@@ -0,0 +1,373 @@
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+#include <stdint.h>
+#include <string.h>
+
+#include "localeInformationConstants.hpp"
+
+#include "unicode/dcfmtsym.h" //decimal
+#include "unicode/dtfmtsym.h" //date
+#include "unicode/ulocdata.h"
+
+/*
+Function:
+GetLocale
+
+Returns a locale given the locale name
+*/
+Locale GetLocale(const UChar* localeName, bool canonize = false)
+{
+	char localeNameTemp[ULOC_FULLNAME_CAPACITY];
+
+	if (localeName != NULL)
+	{
+		int32_t len = u_strlen(localeName);
+		u_UCharsToChars(localeName, localeNameTemp, len);
+		localeNameTemp[len] = (char)0;
+	}
+
+	Locale loc;
+	if (canonize)
+	{
+		loc = Locale::createCanonical(localeName == NULL ? NULL : localeNameTemp);
+	}
+	else
+	{
+		loc = Locale::createFromName(localeName == NULL ? NULL : localeNameTemp);
+	}
+
+	return loc;
+}
+
+/*
+Function:
+GopyCharToUChar
+
+Copies the given char to UChar with error checking
+*/
+UErrorCode CopyCharToUChar(const char *str, UChar* value, int32_t valueLength)
```

Comment:
I would give this a a "worse" name, given that u_charsToUChars() can't take any char* as input.  maybe u_charsToUChars_safe?